GetRecd/Services/AuthService.swift

```diff
@@ -13,11 +13,52 @@ import FirebaseCore
 import FirebaseAuth
 import GoogleSignIn
 
-class AuthService {
+class AuthService: NSObject, GIDSignInDelegate {
     // Static variable used to call AuthService functions
     static let instance = AuthService()
     private var authInstance: Auth?
 
+    private override init() {
+        super.init()
+        GIDSignIn.sharedInstance().delegate = self
+        GIDSignIn.sharedInstance().clientID = FirebaseApp.app()?.options.clientID
+    }
+
+    func sign(_ signIn: GIDSignIn!, didSignInFor user: GIDGoogleUser!, withError error: Error?) {
+        if let error = error {
+            print(error.localizedDescription)
+            return
+        }
+
+        guard let authentication = user.authentication else { return }
+        let credential = GoogleAuthProvider.credential(withIDToken: authentication.idToken,
+                accessToken: authentication.accessToken)
+
+        Auth.auth().signIn(with: credential) { (user, error) in
+            if let error = error {
+                print(error.localizedDescription)
+                return
+            }
+            var userData = [String:Any]()
+            if (user!.displayName != nil) {
```

Comment:
We really shouldn't be force unwrapping the user here. I would recommend doing an if let or guard statement to avoid crashes with no diagnostic.